src/lib/x509/certstor.h

```diff
@@ -31,6 +31,15 @@ class BOTAN_DLL Certificate_Store
          find_cert(const X509_DN& subject_dn, const std::vector<byte>& key_id) const = 0;
 
       /**
+      * Find a certificate by searching for one with a matching SHA-1 hash of
+      * public key. Used for OCSP.
+      * @param key_hash SHA-1 hash of the subject's public key
+      * @return a matching certificate or nullptr otherwise
+      */
+      virtual std::shared_ptr<const X509_Certificate>
+         find_cert_by_pubkey_sha1(const std::vector<byte>& key_hash) const = 0;
```

Comment:
There are some things we can do to improve things in the near term, even if we can't remove SHA-1 entirely from the build.

- [easy] Change default path validation parameters to ~110 expected strength instead of 80. This disables SHA-1 as a trusted hash during certificate validation, and increases RSA minimum to (just under) 2048 bit keys, instead of 1024 allowed now.

- [easy] Disable SHA-1 HMAC in TLS by default. Not sure if it's a real issue

- [medium] Restructure TLS code to allow disabling MD5/SHA-1 (and coorespondingly disabling TLS v1.0/v1.1 support, since these versions require MD5+SHA-1)